File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long a=Math.abs(x);
        long r=0;
        while(a>0){
            long k=a%10;
            r=r*10+k;
            a/=10;
        }
        if(r>Integer.MAX_VALUE || r<Integer.MIN_VALUE){
            return 0;
        }
        if(x>0){
            return (int)r;
        }
        else{
            return (int)(-1*r);
        }
        
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:

    def reverse(self, x: int) -> int:

        temp = abs(x)

        res = 0

        while temp > 0:

            digit = temp % 10

            res = res * 10 + digit

            temp = temp // 10

        if res > 2**31 - 1 or res < -2**31:

            return 0

        if x > 0:

            return res

        else:

            return -res